src/app/child-dev-project/children/demo-data-generators/demo-child-generator.service.ts

```diff
@@ -72,6 +77,49 @@ export class DemoChildGenerator extends DemoDataGenerator<Child> {
     for (let i = 1; i <= this.config.count; i++) {
       data.push(DemoChildGenerator.generateEntity(String(i)));
     }
+    // this.addFiles(data);
     return data;
   }
+
+  async addFiles(children: Child[]) {
+    const fileNames = await fetch("assets/data/files.txt")
+      .then((res) => res.text())
+      .then((res) => res.split("\n"));
+    let fileNum = 0;
+    for (let child of children) {
+      for (let j = 1; j < 12; j++) {
+        const fileName = fileNames[fileNum];
+        const type = this.getContentType(fileName);
+        await fetch(`assets/data/${fileName}`)
+          .then((res) => res.blob())
+          .then((file) =>
+            firstValueFrom(
+              this.fileService.uploadFile(
+                { type } as File,
+                child._id,
+                `file${j.toString()}`,
+                file
+              )
+            )
+          );
+        console.log("uploaded file", fileNum, fileName);
+        fileNum = (fileNum + 1) % 1064;
+      }
+    }
+    console.log("done");
```

Comment:
Ahem, this will be removed 😅 I used this code for creating my demo data. At the moment there is no concept on how this would work without a real database in the backend (which is the case in the demo mode)

Reply:
Demo functionality is now here through the `MockFileService`